pytest/tests/sanity/rosetta.py

```diff
@@ -0,0 +1,305 @@
+#!/usr/bin/env python3
+
+import dataclasses
+import json
+import os
+import pathlib
+import sys
+import time
+import typing
+import unittest
+
+import base58
+import ed25519
+import requests
+
+sys.path.append(str(pathlib.Path(__file__).resolve().parents[2] / 'lib'))
```

Comment:
what is this magic

Reply:
It figures out path to the `pytest/lib` directory and adds it to Python’s search path so modules from there can be imported.

`__file__` is path to the current file. `pathlib.Path(__file__).resolve()` makes in an absolute path. `parents[2]` returns it’s great-grandparent; that gives us path to `pytest` directory. `/ 'lib'` joins that with `lib` which is the directory we’re after.

Reply:
I see....